app/pages/lab/iiif/components/helpers/parseManifestV2.js

```diff
@@ -24,11 +24,21 @@ function parseCanvas(canvas, index) {
 }
 
 function parseValue({ label, value }) {
-  if (value[`@language`]) {
-    const language = value[`@language`];
+  let language
+
+  if (value['@language']) {
+    language = value['@language'];
     label = `${label}:${language}`;
   }
 
+  if (label['@language']) {
+    language = label['@language'];
+    label = `${label['@value']}:${language}`;
+    if (Array.isArray(value)) {
+      value = value.find(v => v['@language'] === language)
+    }
+  }
+
   if (value[`@value`]) {
```

Comment:
[very minor] I realise you've been changing the tildes (`) to (') in things like `value['@language']` above, (makes sense) and I assume you'd want to change this line as well.

Reply:
Good catch. I’ve been changing these to plain strings because they don’t need to be evaluated as templates.